util/execdetails/execdetails.go

```diff
@@ -141,6 +157,42 @@ func (e *RuntimeStatsColl) Get(planID string) *RuntimeStats {
 	return runtimeStats
 }
 
+// GetCop gets a CopRuntimeStats by planID and address.
+func (e *RuntimeStatsColl) GetCop(planID, address string) *CopRuntimeStats {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	stats, ok := e.copStats[planID]
+	if !ok {
+		stats = make(map[string]*CopRuntimeStats, 8)
+		e.copStats[planID] = stats
+	}
+	stat, ok := stats[address]
+	if !ok {
+		stat = new(CopRuntimeStats)
+		stats[address] = stat
+	}
+	return stat
+}
+
+// CopSummary gets a summary of the cop task's execution information.
+func (e *RuntimeStatsColl) CopSummary(planID string) string {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	stats, ok := e.copStats[planID]
+	if !ok {
+		return ""
+	}
+	var maxProcesses, totalRows, totalIters uint64
+	for _, stat := range stats {
+		if stat.timeProcessedNs > maxProcesses {
+			maxProcesses = stat.timeProcessedNs
+		}
+		totalRows += stat.numProducedRows
+		totalIters += stat.numIterations
+	}
+	return fmt.Sprintf("max proc ns:%v, total rows:%v, total iters:%v", maxProcesses, totalRows, totalIters)
```

Comment:
Can you explain more why we use the max execution times in a tikv? Sometimes, several cop tasks can be executed parallelly in one tikv.

Reply:
What I concern is that it is the sum of execution time in a tikv, and two cop tasks may execute parallelly on a tikv, so the time may miss lead us.

Reply:
I just get what you mean.. It's really a problem.. I will fix it.